src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/LimitedUseItem.java

```diff
@@ -0,0 +1,156 @@
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+
+import io.github.thebusybiscuit.cscorelib2.chat.ChatColors;
+import io.github.thebusybiscuit.slimefun4.api.SlimefunAddon;
+import io.github.thebusybiscuit.slimefun4.api.items.ItemState;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.magical.staves.StormStaff;
+import io.github.thebusybiscuit.slimefun4.utils.LoreBuilder;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This class represents an item with a limited number of uses.
+ * When the item runs out of "uses", it breaks.
+ *
+ * @author Linox
+ * @author Walshy
+ * @author TheBusyBiscuit
+ * @author martinbrom
+ *
+ * @see StormStaff
+ */
+public abstract class LimitedUseItem extends SimpleSlimefunItem<ItemUseHandler> {
+
+    private static final NamespacedKey usageKey = new NamespacedKey(SlimefunPlugin.instance(), "uses_left");
+    private static final Pattern REGEX = Pattern.compile(ChatColors.color("&e[0-9]+ Uses &7left"));
+
+    private int maxUseCount = -1;
+
+    @ParametersAreNonnullByDefault
+    public LimitedUseItem(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
+        super(category, item, recipeType, recipe, null);
+
+        addItemHandler(getItemHandler());
+    }
+
+    /**
+     * Returns the number of times this item can be used.
+     *
+     * @return The number of times this item can be used.
+     */
+    public final int getMaxUseCount() {
+        return maxUseCount;
+    }
+
+    /**
+     * Sets the maximum number of times this item can be used.
+     * The number must be greater than zero.
+     *
+     * @param count The maximum number of times this item can be used.
+     * @return The {@link LimitedUseItem} for chaining of setters
+     */
+    public final @Nonnull LimitedUseItem setMaxUseCount(int count) {
+        Validate.isTrue(count > 0, "The maximum use count must be greater than zero!");
+
+        if (getState() == ItemState.UNREGISTERED) {
+            maxUseCount = count;
+            return this;
+        } else {
+            throw new IllegalStateException("You cannot modify the maximum use count after the Item was registered.");
```

Comment:
ummm, why not? 🤔

Reply:
I feel like that's just waiting for problems to happen when we start modifying 'constant' values during runtime. I can, of course, remove this but the same check exists for energy capacity, energy consumption, and other values.

Reply:
Energy values are a different matter though, these get passed to the energy network upon registration, so any change to them would need to be synchronized to the entire energy network handling and be dealt with.
Here I see rather few problems that would arise from this as long as it is kept as a positive integer. But I am fine with the warning here at first, as long as it is properly justified and documented.